mathml/relations/css-styling/first-line-first-letter-pseudo-elements-003.html

```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>MathML and ::first-line/::first-letter pseudo-elements</title>
+<link rel="help" href="https://w3c.github.io/mathml-core/#legacy-mathml-style-attributes">
+<link rel="match" href="first-line-first-letter-pseudo-elements-003-ref.html"/>
+<meta name="assert" content="display math elements do not contribute a first formatted line/letter to ancestors.">
+<style>
+  .redfirstline::first-line { background: red; }
+  .redfirstletter::first-letter { background: red; }
+</style>
+<p>PASS if there is no red.</p>
+<ol>
+  <li>
+    <div class="redfirstline">
+      <math style="display: inline math"><mtext>Hello<br>World!</mtext></math>
+    </div>
+  </li>
+  <li>
+    <math class="redfirstletter">
+      <mtext style="display: inline math">Hello<br>World!</mtext>
+    </math>
+  </li>
+  <li>
+    <div class="redfirstline">
+      <math class="redfirstletter">
+        <mtext style="display: block math">Hello<br>World!</mtext>
```

Comment:
I think rather than setting the "display" on `mtext`, you could put everything on the `<math>` tag i.e.

`<math style="display: inline math">` and `<math style="display: block math">`

or with equivalent MathML-only code

`<math>` and  `<math display="block">`.

Reply:
Could or should? Is there a reason it should matter?

Reply:
I don't remember exactly, but I believe only the `<math>` tag has different rendering for block/inline so it seems relevant to test. And we have the `<math>` and `<math display="block">` syntax which is nice to test with browsers that don't support `display: math`. So that's why i suggested that.

Perhaps you can add tests for the different cases:
- mtext style="display: inline/block math"
- math style="display: inline/block math"
- math display="inline/block"

so we are sure we cover the different cases?

Reply:
created https://github.com/w3c/mathml-core/issues/236 for that